pkg/event/sqlite.go

```diff
@@ -273,8 +289,12 @@ func (repo SQLiteRepository) ListAccessEvents(ctx context.Context, listParams Li
 		CreatedAt: lastAccessEvent.CreatedAt,
 	})
 	if err != nil {
-		return accessEvents, "", err
+		return models, "", err
+	}
+
+	for i := range accessEvents {
```

Comment:
This loop should be moved above line 282 so that in the scenario that the result set is smaller than the specified `Limit`, the appropriate result is still returned.